identity-iota/src/tangle/encoding.rs

```diff
@@ -0,0 +1,54 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::error::Error::CompressionError;
+use crate::tangle::compression_brotli::compress_brotli;
+use crate::tangle::compression_brotli::decompress_brotli;
+use crate::Error;
+
+#[derive(Copy, Clone)]
+pub(crate) enum MessageEncoding {
+  Json = 0,
+  JsonBrotli = 1,
+}
```

Comment:
I wonder if we need to separate `MessageEncoding` and `MessageCompression` into two flags since they can be applied separately? That would avoid duplicating the compression options for each encoding (e.g. `Json, JsonBrotli, Cbor, CborBrotli`). Just a thought, I doubt we'll need a different encoding other than JSON, hopefully.

Reply:
If that will be really necessary in the future, the message version can indicate that. For example starting from message version 2, the second byte is for `MesageEncoding` and the third is for `MessageCompression`. But I doubt using many combinations.